devices/YarpRobotLoggerDevice/src/YarpRobotLoggerDevice.cpp

```diff
@@ -1366,115 +1507,135 @@ void YarpRobotLoggerDevice::run()
     {
         std::lock_guard<std::mutex> lock(signal.mutex);
-        yarp::sig::Vector* vector = signal.port.read(false);
-        if (vector != nullptr)
+        yarp::sig::Vector* collection = signal.port.read(false);
+        if (collection != nullptr)
         {
             if (!signal.dataArrived)
             {
-                m_bufferManager.addChannel({signal.signalName, {vector->size(), 1}});
+                initMetadata(signalName, {});
                 signal.dataArrived = true;
             }
-            m_bufferManager.push_back(*vector, time, signal.signalName);
         }
     }
 
-    int bufferportSize = m_textLoggingPort.getPendingReads();
-    BipedalLocomotion::TextLoggingEntry msg;
-
-    while (bufferportSize > 0)
+    // lock guard scope for text logging port
     {
-        yarp::os::Bottle* b = m_textLoggingPort.read(false);
-        if (b != nullptr)
+        std::lock_guard<std::mutex> lockGuardTextLogging(m_textLoggingPortMutex);
+        int bufferportSize = m_textLoggingPort.getPendingReads();
+        BipedalLocomotion::TextLoggingEntry msg;
+
+        while (bufferportSize > 0)
         {
-            msg = BipedalLocomotion::TextLoggingEntry::deserializeMessage(*b, std::to_string(time));
-            if (msg.isValid)
+            yarp::os::Bottle* b = m_textLoggingPort.read(false);
+            if (b != nullptr)
             {
-                signalFullName = msg.portSystem + "::" + msg.portPrefix + "::" + msg.processName
-                                 + "::p" + msg.processPID;
+                msg = BipedalLocomotion::TextLoggingEntry::deserializeMessage(*b, std::to_string(time));
+                if (msg.isValid)
+                {
+                    signalName = msg.portSystem + "::" + msg.portPrefix + "::" + msg.processName + "::p"
+                                + msg.processPID;
 
-                // matlab does not support the character - as a key of a struct
-                findAndReplaceAll(signalFullName, "-", "_");
+                    // matlab does not support the character - as a key of a struct
+                    findAndReplaceAll(signalName, "-", "_");
 
-                // if it is the first time this signal is seen by the device the channel is
-                // added
-                if (m_textLogsStoredInManager.find(signalFullName)
-                    == m_textLogsStoredInManager.end())
-                {
-                    m_bufferManager.addChannel({signalFullName, {1, 1}});
-                    m_textLogsStoredInManager.insert(signalFullName);
+                    // if it is the first time this signal is seen by the device the channel is
+                    // added
+                    if (m_textLogsStoredInManager.find(signalName) == m_textLogsStoredInManager.end())
+                    {
+                        m_bufferManager.addChannel({signalName, {1, 1}});
+                        m_textLogsStoredInManager.insert(signalName);
```

Comment:
You can restore the original name of the variable `signalFullName` instead of the new one `signalName`

Reply:
Fixed in [9555d00](https://github.com/ami-iit/bipedal-locomotion-framework/commit/9555d00dcfb5c845166c6600713b601ec7991c96)